bootbox.js

```diff
@@ -410,17 +410,9 @@
     };
 
     options.buttons.confirm.callback = function() {
-      var value;
-
-      if (options.inputType === "checkbox") {
-        value = input.find("input:checked").map(function() {
-          return $(this).val();
-        }).get();
-      } else {
-        value = input.val();
-      }
-
-      return options.callback.call(this, value);
+      // trigger form submit, including native validation
+      $("<input type='submit'>").hide().appendTo(form).click().remove();
```

Comment:
Seems like it would easier to simply trigger the submit event on the form itself. Is there a reason you're adding and removing a submit button?

Reply:
yep. If you call `submit()`, you won't trigger native validation on form and the form will be submitted despite it isn't valid.